network/p2p/libp2pNode.go

```diff
@@ -359,8 +360,25 @@ func (n *Node) tryCreateNewStream(ctx context.Context, identity flow.Identity, m
 		default:
 		}
 
+		// TODO: why were we doing this? Is it okay to remove?
 		// remove the peer from the peer store if present
-		n.host.Peerstore().ClearAddrs(peerID)
+		// n.host.Peerstore().ClearAddrs(peerID)
```

Comment:
@vishalchangrani  ?

Reply:
this was done to avoid libp2p doing any exponential backoffs for a connection it tried earlier and failed but still has an entry for it in the peer store.
I can check if it is still needed or not.

Reply:
Ahh I see... Okay let me know.